### PR TITLE
feat(components): add box primitive component

### PR DESCRIPTION
**Pull request by richbachman**
## Description of the change

Screen Shot 2022-09-19 at 09 35 28

This is the first version of the Box primitive component. This component is a utility component that we can use to build other. components in the system. It's a mix of styled system concepts and css prop that ties into our theme config.

Box comes with props that pull styles directly from the Pluto theme, i.e.: `backgroundColor`. Responsive values can also be included in the props, i.e.: 

```
backgroundColor: {
    "@initial": "colorBackgroundWeak",
    "@media (min-width: 800px)": "colorBackgroundInfo",
},
```

The `css` prop can also be used to styled the Box in using any valid css rules. The `css` prop also ties directly to the Pluto theme, and can accept responsive and psuedo element values, i.e.:

```
css={{
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    "&:hover": {
        cursor: "pointer",
        boxShadow: "$shadowStrong",
    },
}}
```

There are also some minor plop template edits as well as build config edits in this PR.

## Testing the change

- [ ] Fire up Storybook and verify the stories display properly
- [ ] Build something in Storybook using Box

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Non-Breaking Change (change to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] Issue from Linear has a link to this pull request

### CODE REVIEW

File: packages/components/src/index.ts [hunk 1]
```diff
@@ -1 +1 @@
-export * from "./components/Sample";
+export * from "./primitives/Box";
```
Comment (apvale) [inline packages/components/src/index.ts:1]: Why are we using the `primitives` concept here? What components should be placed in this folder, for example? 🤔

File: packages/components/src/primitives/Box/Box.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  it("should render with a background color class", () => {
+    render(<Box backgroundColor="colorBackground">{childString}</Box>);
+    const renderedBox = screen.getByText(childString);
+    // Using toHaveAttribute here because the Stitches classNames are dynamic. We just want part of the className.
+    // eslint-disable-next-line jest-dom/prefer-to-have-class
+    expect(renderedBox).toHaveAttribute(
+      "class",
+      expect.stringContaining("backgroundColor-colorBackground")
+    );
```
Comment (apvale) [inline packages/components/src/primitives/Box/Box.test.tsx:28]: So this `expect(renderedBox).toHaveClass("backgroundColor-colorBackground");` doesn't work? 🤔

File: packages/components/package.json [hunk 3]
```diff
@@ -19,14 +19,19 @@
...
     "lint": "TIMING=1 eslint src/**/*.ts* --fix",
     "prettier-check": "prettier --check package.json tsconfig.json \"./src/**/*.{tsx,ts,json}\"",
     "test": "jest",
     "tsc": "tsc"
   },
+  "dependencies": {
+    "@localyze-pluto/theme": "*",
+    "@stitches/react": "^1.2.8",
+    "lodash": "^4.17.21"
+  },
```
Comment (apvale) [inline packages/components/package.json:34]: I tried to run the storybook, and I'm always having this error: 

```
ModuleNotFoundError: Module not found: Error: Can't resolve '@localyze-pluto/theme' in '/Users/anavale/src/pluto/packages/components/src/utils'
    at /Users/anavale/src/pluto/node_modules/webpack/lib/Compilation.js:2016:28
    at /Users/anavale/src/pluto/node_modules/webpack/lib/NormalModuleFactory.js:798:13
    at eval (eval at create (/Users/anavale/src/pluto/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
  ```
  
Am I missing running some commands?

**Review by apvale — COMMENTED** [review]
Thank you for adding the first component, Rich! 🙏🏻

File: packages/components/src/primitives/Box/Box.test.tsx (hunk 2 above)
Comment (richbachman) [reply packages/components/src/primitives/Box/Box.test.tsx:28]: @apvale no. `toHaveClass` expects the full className string to match. `stringContaining` looks for a partial string match. The classNames look like `.pluto-c-lesPJm-bAPvzV-backgroundColor-colorBackgroundInfo` where the `c-lesPJm-bAPvzV` part is dynamic.

File: packages/components/src/index.ts (hunk 1 above)
Comment (richbachman) [reply packages/components/src/index.ts:1]: Primitives are the building block components we can use to build other components. Things like Box and Text. We could also have an un-styled collapse, menu, and tab primitives. Those would come with the functionality but not styles. Some of these may just be exported headless ui components.

**Review by apvale — APPROVED** [review]

**Pull request merged**